### PR TITLE
Process instance migration concept documentation

### PR DESCRIPTION
**Pull request by korthout**
## Description

This adds documentation for the new Process Instance Migration concept.

Please have a look, and provide any feedback.

https://www.loom.com/share/627d3d8f1973471eabd3c687c8d154af?sid=8a8e57bf-39c0-4b9e-943e-439ac7031d98

## When should this change go live?

- [ ] This change is not yet live and should not be merged until {ADD_DATE} (apply `hold` label or convert to draft PR)?
- [ ] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [x] This functionality is already available but undocumented (since `8.4.0-alpha2`)
- [ ] This is a bug fix or security concern.

## PR Checklist

- [ ] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [x] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Comment by korthout** [comment]
This is just an initial write-up for the docs. Most screenshots are still missing.

@pedesen @christinaausley @gastonpillet01 @berkaycanbc Please have a look and provide any feedback

File: docs/components/concepts/process-instance-migration.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+However, the requirements of our process have changed.
+
+Instead of completing the process, we want to add a user task `B` after the service task `A` and before the end event.
+
+<!-- TODO: add screenshot of simple example target process with an additional user task `B` after `A` -->
+
+We can create new process instances according to this new process model after deploying it, but we also want our process instance to reach user task `B` when service task `A` is completed.
+
+<!--
+I'm using process, process model, and process definition interchangeably, because I don't know what's best.
```
Comment (christinaausley) [inline docs/components/concepts/process-instance-migration.md:49]: I will create an issue for this to define usage of these terms and adjust the documentation retroactively, as this is likely an issue across more than just this doc.

File: docs/components/concepts/process-instance-migration.md [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+You cannot map an element to an element of a different type.
+An active service task can only be mapped to an active service task.
+It cannot be mapped to a user task as this changes the element type.
+
+Also note that the [jobs, expressions, and input mappings](#jobs-expressions-and-input-mappings) are not recreated.
+So, while service task `C` is active in the target process, the associated job still has the job type from when it was associated to service task `A`.
+:::
+
+<!--
+Should we even have this section 'Changing the active elements' at all?
```
Comment (christinaausley) [inline docs/components/concepts/process-instance-migration.md:93]: I am all about trimming up a doc if the information isn't necessarily beneficial. I would remove this, or just add a note summarizing (which you kind of already do below).

**Review by christinaausley — COMMENTED** [review]
Looks good -- I'll get this approved and merged once you have screenshots ready 👍

**Review by berkaycanbc — DISMISSED** [review]
Amazing work @korthout !!! 🚀 

I really like the examples and remarks for jobs, expressions, timers and other very important details. Very good!🥇

I added few suggestions to better transfer the meaning to the reader. Please feel free to use suggestions I provided as an example and re-write your own sentences. 👍

**Comment by korthout** [comment]
Thanks for the reviews @berkaycanbc and @christinaausley 🙇 

I've applied some changes (adding screenshots) and a few additional links/notes. Please have another look. 🙇 

📢 From this moment onwards, I'd like to hand over the PR to @berkaycanbc as I'm taking FTO.

Some last things to note:
- We can remove the danger banner at the top as soon as we resolve: camunda/zeebe/issues/15129 
- Near the top of the page is a `note` admonition that navigates readers to Operate's user guide for process instance migration. It's not yet available, but the link has to be adjusted when it does become available. You can track that documentation in: https://github.com/camunda/camunda-docs/issues/2848
- Please consider any suggestions shared by @gastonpillet01 or @pedesen 
- I believe that, even in its current form, it could already be published as part of `8.4.0-alpha2` docs

**Comment by christinaausley** [comment]
> Thanks, @korthout, for wrapping up the PR documentation. Let's keep this open for now to decide later if we keep the "danger" section.

@abbasadel Any additional comments here? If not, planning to merge today 👍

**Comment by christinaausley** [comment]
@berkaycanbc Any additional changes here?

**Review by akeller — APPROVED** [review]
Approving to merge since this is tagged for alpha2. If more changes need to be made, please open a new PR.

**Comment by berkaycanbc** [comment]
> Approving to merge since this is tagged for alpha2. If more changes need to be made, please open a new PR.

@akeller I will open another PR for additional changes. 👍

**Pull request merged**